packages/tests/bin/runMochaLayers.sh

```diff
@@ -34,15 +34,16 @@ else
 fi
 
 if [ -z "$API_HOST" ]; then
-    . ~/.wskprops
-    export API_HOST=$APIHOST
-fi
-
-if [ ! -f ~/.wskprops ]; then
+    if [ -f ~/.wskprops ]; then
+        . ~/.wskprops
+        export API_HOST=$APIHOST
+    fi
+else
     echo "APIHOST=$API_HOST" > ~/.wskprops
```

Comment:
Thanks! Your change looks good for openwhisk developers. If the developers didn't set the openwhisk configuration properly, the test used to fail before mocha starts. With your change, electron openwhisk tests will fail with `authentication error` after trying real openwhisk commands, and webpack openwhisk  tests will fail with `apihost error` at the mocha `before hook`, which look good to me!

It'd be nice if we can also let non-openwhisk developers use `npm test` locally. With your current change, non-openwhisk users will still be treated as openwhisk developers and get authentication errors as above. I think openwhisk tests should be skipped in this case. Currently. we have two environment variables to do the test filtering. The first one is LAYERS for testing against specific test suites, and the other one is TEST_FILTER for testing against specific test case. 

Example Usage:
 ```LAYERS="core bash-like openwhisk1" npm run test```
  ```TEST_FILTER="shell commands" npm run test```

(TEST_FILTER is how we use `mocha filter`, you can find what to filter in the `describe()` line of each test.)

Instead of forcing developers to use the above environment variables, it would be nice if we could guess the developers's intention. When they do `npm test` locally without openwhisk configuration, we should filter out openwhisk test suites for them (or skip the tests in mocha `before hook` if it's easy to implement). In this case, we should also give them warning/info that the openwhisk tests have been skipped for the lack of configuration, credentials, etc. However, if `npm test` happened on travis, (we do this exactly for travis headless tests), we shouldn't filter out any tests due to the lack of proper configuration, because travis is intentional to run all the tests. 

We should also consider replacing the current `npm test` with a quick sanity check which is short, straightforward, and doesn't require any credentials. 🤔

Reply:
Sure, I have still not finished the PR. I should have marked this as a draft. I want to pass the commented error before sending more changes. :smile: